### PR TITLE
Reverse order of plugin name and widget name

### PR DESCRIPTION
**Pull request by mstabrin**
# Description

After playing around with tabified dock widgets the users love it very much, but find it hard to find the actual tab.
This is, because the current dock widget naming is `plugin_name: widget_name`.
Here, I propose that the naming is inverted to `widget_name: plugin_name` since the widget_name seems to be much more important to the user for their work.

Additionally, the first dock widget should be selected by default rather than the last to be consistent with the order in the `Plugins` menu and the behavior when no `widget_name` is provided via the CLI.  

Example:
`napari -w napari-boxmanager __all__`
plugin_name: `napari-boxmanager`
widget_names: `'add_layer', 'match_mics', 'bandpass_filter', 'boxmanager'`

Before:
image

Now:
image

## Type of change
- [X] Bug-fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by brisvag** [comment]
Good idea! Totally agree on the order. I suggest using parenthesis or something like that for the plugin name; I find that this conveys the "hiearachy" better: `widget-name (plugin)`.

Alternatives:
- put the name of the plugin in a frame *around* the tabs (maybe unfeasible with our current dock widget stuff
- do not put the plugin name in the tab, but only in a hover tooltip.
- scrolling tab headers?

**Comment by mstabrin** [comment]
@brisvag I like the parenthesis approach :) 
The name of the tab seems to be deeply connected to the menu functionality, so the alternatives seem to be quite complicated for now. Not sure how to allow for scrolling tab headers, but I am also not sure how much they would contribute to the user experience :)

**Comment by mstabrin** [comment]
I added a new CLI command which allows for the print of all available plugins:

```
- napari -wh                
Available plugins

TYPE: dock - NPE2
  PLUGIN_NAME: 'napari-matplotlib'
    WIDGET_NAME: 'Histogram'
    WIDGET_NAME: 'Scatter'
    WIDGET_NAME: 'FeaturesScatter'
    WIDGET_NAME: '1D slice'
  PLUGIN_NAME: 'napari-boxmanager'
    WIDGET_NAME: 'bandpass_filter'
    WIDGET_NAME: 'match_mics'
    WIDGET_NAME: 'add_layer'
    WIDGET_NAME: 'boxmanager'

TYPE: dock - PLUGIN_MANAGER
  PLUGIN_NAME: 'Image-Composer'
    WIDGET_NAME: 'Composer'
  PLUGIN_NAME: 'napari-mrcfile_handler'
    WIDGET_NAME: 'Pixel Spacing'
```

I find it quite helpful to build my `napari -w` command

File: napari/__main__.py
Comment (Czaki) [inline napari/__main__.py]: here is bug
```suggestion
        shared_dict.setdefault(_key, {}).setdefault(
```

**Review by Czaki — CHANGES_REQUESTED** [review]
I try to test these changes locally and get:
```
$ napari -wh               
Traceback (most recent call last):
  File "/home/czaki/.pyenv/versions/napari/bin/napari", line 8, in <module>
    sys.exit(main())
  File "/home/czaki/Projekty/napari/napari/__main__.py", line 600, in main
    _run()
  File "/home/czaki/Projekty/napari/napari/__main__.py", line 261, in _run
    args, kwargs = parse_sys_argv()
  File "/home/czaki/Projekty/napari/napari/__main__.py", line 222, in parse_sys_argv
    show_plugin_info()
  File "/home/czaki/Projekty/napari/napari/__main__.py", line 242, in show_plugin_info
    shared_dict.setdefault(_key, []).setdefault(
AttributeError: 'list' object has no attribute 'setdefault'  
```

Also wh should be described in first comment (you could edit it) with the description of `wh` and function `show_plugin_info` should be called in tests.

File: napari/__main__.py
Comment (mstabrin) [reply napari/__main__.py]: Removed printing of plugins from this PR -> Will move it to a separate one :)

**Review by Czaki — APPROVED** [review]
Looks ok to me. Thanks. Let's wait for 24h to allow another person to review.

File: napari/plugins/__init__.py
Comment (andy-sweet) [inline napari/plugins/__init__.py]:
```suggestion
# widget_name (plugin_name)
```

**Comment by Czaki** [comment]
@mstabrin There is no need to update after each PR is merged to the main branch.

**Pull request merged**
